tools/webdriver/webdriver/error.py

```diff
@@ -213,6 +221,8 @@ def from_response(response):
     code = value["error"]
     message = value["message"]
     stack = value["stacktrace"] or None
+    # data is optional, and could even be an empty dict
+    data = value.get("data") or None
```

Comment:
This was deliberate, like the `value["stacktrace"] or None` above — while `value.get("data")` returns `None` when `"data" not in value`, it doesn't when `data.get("value")` returns `{}`.

i.e., it's canonicalising empty values to `None`; we should either canonicalise all empty values to `None` or… none. (albeit noting that error is special here and probably shouldn't be!)

Reply:
I think we shouldn’t modify the returned value if it’s an empty list (or any similar value that evaluates to `None`). Doing so would alter the data we actually want to test and could lead to discrepancies with the spec—or could let issues slip by unnoticed, such as when a client returns an incorrect type.

That said, as you pointed out, this approach currently aligns with how `stacktrace` is handled, so it may be acceptable for now. However, it would be a good idea to open a follow-up issue to discuss the potential removal of this fallback.